whole_body_controllers/include/whole_body_controllers/ik_controller.hpp

```diff
@@ -0,0 +1,117 @@
+
+// auto-generated by generate_parameter_library
+#include <whole_body_controllers/ik_controller_parameters.hpp>
+
+namespace whole_body_controllers
+{
+
+class IKController : public controller_interface::ChainableControllerInterface
+{
+public:
+  IKController() = default;
+
+  auto on_init() -> controller_interface::CallbackReturn override;
+
+  // NOLINTNEXTLINE(modernize-use-nodiscard)
+  auto command_interface_configuration() const -> controller_interface::InterfaceConfiguration override;
+
+  // NOLINTNEXTLINE(modernize-use-nodiscard)
+  auto state_interface_configuration() const -> controller_interface::InterfaceConfiguration override;
+
+  auto on_configure(const rclcpp_lifecycle::State & previous_state) -> controller_interface::CallbackReturn override;
+
+  auto on_activate(const rclcpp_lifecycle::State & previous_state) -> controller_interface::CallbackReturn override;
+
+  auto update_and_write_commands(const rclcpp::Time & time, const rclcpp::Duration & period)
+    -> controller_interface::return_type override;
+
+protected:
+  auto on_export_reference_interfaces() -> std::vector<hardware_interface::CommandInterface> override;
+
+  auto update_reference_from_subscribers(const rclcpp::Time & time, const rclcpp::Duration & period)
+    -> controller_interface::return_type override;
+
+  auto update_system_state_values() -> controller_interface::return_type;
+
+  auto update_parameters() -> void;
+
+  auto configure_parameters() -> controller_interface::CallbackReturn;
+
+  auto update_and_validate_interfaces() -> controller_interface::return_type;
+
+  std::shared_ptr<pinocchio::Model> model_;
+  std::shared_ptr<pinocchio::Data> data_;
+
+  realtime_tools::RealtimeBuffer<geometry_msgs::msg::Pose> reference_;
+  std::shared_ptr<rclcpp::Subscription<geometry_msgs::msg::Pose>> reference_sub_;
+
+  // allow users to send the vehicle state as a message - this can simplify integration with state estimators
+  // that publish the vehicle state without the manipulator states
+  std::shared_ptr<rclcpp::Subscription<nav_msgs::msg::Odometry>> vehicle_state_sub_;
+  realtime_tools::RealtimeBuffer<nav_msgs::msg::Odometry> vehicle_state_;
+  std::vector<double> position_state_values_, velocity_state_values_;
+
+  std::unique_ptr<pluginlib::ClassLoader<ik_solvers::IKSolver>> loader_;
+  std::shared_ptr<ik_solvers::IKSolver> solver_;
+
+  std::unique_ptr<ik_controller::ParamListener> param_listener_;
+  ik_controller::Params params_;
+
+  std::shared_ptr<rclcpp::Publisher<auv_control_msgs::msg::IKControllerStateStamped>> controller_state_pub_;
+  std::unique_ptr<realtime_tools::RealtimePublisher<auv_control_msgs::msg::IKControllerStateStamped>>
+    rt_controller_state_pub_;
+
+  // make the free-flyer position and velocity dof names "static"
+  std::vector<std::string> free_flyer_pos_dofs_{"x", "y", "z", "qx", "qy", "qz", "qw"};
+  std::vector<std::string> free_flyer_vel_dofs_{"x", "y", "z", "rx", "ry", "rz"};
```

Comment:
It might be worth making these configurable similar to the other controllers. This could be something like "position_joints" and "velocity_joints"